pkg/controller/volume/persistentvolume/index.go

```diff
@@ -117,9 +117,14 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *api.PersistentVo
 		//   the claim.
 		for _, volume := range volumes {
 			if isVolumeBoundToClaim(volume, claim) {
-				// this claim and volume are bound; return it,
-				// whether the claim is prebound or for volumes
-				// intended for dynamic provisioning v1
+				// this claim and volume are pre-bound; return
+				// the volume if the size request is satisfied,
+				// otherwise leave the claim pending
+				volumeQty := volume.Spec.Capacity[api.ResourceStorage]
+				volumeSize := volumeQty.Value()
+				if volumeSize < requestedSize {
+					return nil, nil
```

Comment:
If the pre-bound PV is too small we should check all the other volumes and not blindly return here. There may be other PVs that could satisfy the claim.

`continue` would be more appropriate.